### PR TITLE
Add reward testing workflow

### PR DESCRIPTION
**Pull request by ManuelActisCassa**
Tests integrated with github workflow.

For now, implemented tests for velocity baseline rewards.
1 test function for each reward, but function may contain many test cases.

To run tests : 
uv run --extra cpu pytest

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+  Returns:
+    XML content as string.
+  """
+  fixtures_path = Path(__file__).parent / "fixtures"
+  if not fixture_name.endswith(".xml"):
+    fixture_name = f"{fixture_name}.xml"
+  fixture_file = fixtures_path / fixture_name
+  return fixture_file.read_text()
+
```
Comment (saikishor) [inline tests/conftest.py]: We don't have any right?

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    actuator_cfg: Optional actuator configuration.
+
+  Returns:
+    Entity instance.
+  """
+  xml_string = load_fixture_xml(fixture_name)
+  if actuator_cfg is not None:
+    return create_entity_with_actuator(xml_string, actuator_cfg)
+  cfg = EntityCfg(spec_fn=lambda: mujoco.MjSpec.from_string(xml_string))
+  return Entity(cfg)
```
Comment (saikishor) [inline tests/conftest.py]:
```suggestion
```

File: tests/conftest.py (hunk 1 above)
Comment (saikishor) [reply tests/conftest.py]:
```suggestion
```

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+def floating_base_articulated_xml() -> str:
+  """Load floating base articulated robot XML fixture."""
+  return load_fixture_xml("floating_base_articulated")
+
+
+@pytest.fixture
+def biped_xml() -> str:
+  """Load biped robot XML fixture with ground plane."""
+  return load_fixture_xml("biped")
+
```
Comment (saikishor) [inline tests/conftest.py]:
```suggestion
```

**Review by saikishor — CHANGES_REQUESTED** [review]

File: tests/test_kangaroo_velocity_rewards.py [hunk 4]
```diff
@@ -0,0 +1,976 @@
...
+  env = mock_env
+
+  mock_asset_cfg.joint_ids = [0, 1, 2]
+
+  asset = env.scene[mock_asset_cfg.name]
+
+  asset.data.joint_pos = torch.zeros((env.num_envs, 3), device=env.device)
+  asset.data.soft_joint_pos_limits = torch.ones((env.num_envs, 3, 2), device=env.device)
+  asset.data.soft_joint_pos_limits[:, :, 0] *= -1.0
+
```
Comment (saikishor) [inline tests/test_kangaroo_velocity_rewards.py:447]:
```suggestion

  value = mjlab_r.joint_pos_limits(env, mock_asset_cfg)

  test_name = "Dofs pos limits"

  assert value.shape == (env.num_envs,), tensor_shape_error_message(
    test_name, (env.num_envs,), value.shape
  )
  assert value[0] == pytest.approx(0.0, abs=1e-6), tensor_value_error_message(test_name)
  
```

File: tests/test_kangaroo_velocity_rewards.py [hunk 5]
```diff
@@ -0,0 +1,976 @@
...
+  assert value.shape == (env.num_envs,), tensor_shape_error_message(
+    test_name, (env.num_envs,), value.shape
+  )
+  assert value[0] == pytest.approx(0.04, abs=1e-6), tensor_value_error_message(
+    test_name
+  )
+
+
+def test_soft_lanfing_penalty(mock_env):
```
Comment (saikishor) [inline tests/test_kangaroo_velocity_rewards.py]:
```suggestion
def test_soft_landing_penalty(mock_env):
```

**Review by saikishor — CHANGES_REQUESTED** [review]

**Review by saikishor — APPROVED** [review]
